scripts/cleanup_old_vms.py

```diff
@@ -55,51 +68,59 @@ def match(matchers, vm_name):
         return False
 
 
-def get_vm_config_modified_time(name, vm_name, datastore_url, provider_key):
+def get_vm_config_modified_time(host_name, vm_name, datastore_url, provider_key):
+    """
+    SSH to provider and find modified date for VMX file
+    :param host_name: string the host to connect to
+    :param vm_name: string name of the vm
+    :param datastore_url: string datastore url in format ds:
+    :param provider_key: string provider key
+    :return:
+    """
     try:
         providers_data = cfme_data.get("management_systems", {})
-        hosts = providers_data[provider_key]['hosts']
-        host_creds = providers_data[provider_key].get('host_credentials', 'host_default')
-        hostname = [host['name'] for host in hosts if name in host['name']]
-        if not hostname:
-            hostname = re.findall(r'[0-9]+(?:\.[0-9]+){3}', name)
+        yaml_hosts = providers_data[provider_key]['hosts']
+        # pull host name and credentials from yaml, may be multiple hosts defined
+        hostname, host_cred_key = None, None  # Init in case not found in yaml
+        for host in yaml_hosts:
+            # Some of the vsphere environments are configured with IP as the host's name
+            if host_name in [host['name'], host.get('ipaddress')]:
+                hostname, host_cred_key = host['name'], host['credentials']
+                break
+
+        if not hostname or not host_cred_key:
+            raise KeyError('Could not find host {} and credentials for {} in yaml'
+                           .format(host_name, provider_key))
         connect_kwargs = {
-            'username': credentials[host_creds]['username'],
-            'password': credentials[host_creds]['password'],
-            'hostname': hostname[0]
+            'username': credentials[host_cred_key]['username'],
+            'password': credentials[host_cred_key]['password'],
+            'hostname': hostname
         }
         datastore_path = re.findall(r'([^ds:`/*].*)', str(datastore_url))
-        ssh_client = SSHClient(**connect_kwargs)
-        command = 'find ~/{}/{} -name {} | xargs  date -r'.format(
-            datastore_path[0], str(vm_name), str(vm_name) + '.vmx')
-        exit_status, output = ssh_client.run_command(command)
-        ssh_client.close()
-        modified_time = parser.parse(output.rstrip())
-        modified_time = modified_time.astimezone(pytz.timezone(str(get_localzone())))
-        return modified_time.replace(tzinfo=None)
+        command = 'find ~/{} -name {}.vmx | xargs  date -r'.format(
+            pathjoin(datastore_path[0], vm_name),
+            vm_name)
+
+        with SSHClient(**connect_kwargs) as ssh_client:
+            result = ssh_client.run_command(command)
+        parsed_time = parser.parse(str(result).rstrip())
```

Comment:
Agreed on both counts here, though I'm not too worried about getting more than 1 row since there should only be 1 vmx file here.

Since mgmtsystem doesn't return consistent timestamp formatting for vm_creation_time, I think I would like to leave refactoring the timestamp/age handling until wrapanapi issue #133 is fixed:
https://github.com/ManageIQ/wrapanapi/issues/133